src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Input/Holding_Tests.cs

```diff
@@ -0,0 +1,269 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using FluentAssertions;
+using NUnit.Framework;
+using SamplesApp.UITests.TestFramework;
+using Uno.UITest.Helpers;
+using Uno.UITest.Helpers.Queries;
+using Uno.UITests.Helpers;
+
+namespace SamplesApp.UITests.Windows_UI_Xaml_Input
+{
+	public class Holding_Tests : SampleControlUITestBase
+	{
+		private const string _xamlTestPage = "UITests.Shared.Windows_UI_Input.GestureRecognizerTests.HoldingTests";
+
+		[Test]
+		[AutoRetry]
+		[ActivePlatforms(Platform.Android, Platform.iOS)] // No WASM as Holding is not raise for mouse pointers
+		public void When_Basic()
+		{
+			Run(_xamlTestPage);
+
+			const string targetName = "Basic_Target";
+			const int tapX = 10, tapY = 10;
+
+			// Tap and hold the target
+			var target = _app.WaitForElement(targetName).Single().Rect;
+			Hold(target.X + tapX, target.Y + tapY);
+
+			var result = GestureResult.Get(_app.Marked("LastHeld"));
+			result.Element.Should().Be(targetName);
+			result.State.Should().Be("Completed");
+			((int)result.X).Should().Be(tapX);
+			((int)result.Y).Should().Be(tapY);
+		}
+
+		[Test]
+		[AutoRetry]
+		[Ignore("Currently there is no available gesture to test cancel")]
```

Comment:
Issue to track ?

Reply:
https://github.com/unoplatform/Uno.UITest/issues/31